### PR TITLE
Allow custom getSignedUrl functions

### PR DESCRIPTION
**Pull request by jakubrohleder**
PR addressing odysseyscience/react-s3-uploader#22. It enables to use custom routine to getSignedUrl from server.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,47 @@ The resulting DOM is essentially:
 When a file is chosen, it will immediately be uploaded to S3.  You can listen for progress (and
 create a status bar, for example) by providing an `onProgress` function to the component.
 
+Using custom function to get signedUrl
+------------
+
+If can use custom function to get provide `signedUrl` direclty to `s3uploader` by adding `getSignedUrl` prop. The function you provide should take `file` and `callback` arguments. Callback should be called with an object containing `signedUrl` key.
```
Comment (seanadkinson) [inline README.md]: Fix typos... 

> You can use custom function to provide `signedUrl` directly to `s3uploader` by adding `getSignedUrl` prop.  The function you provide...

File: README.md [hunk 2]
```diff
@@ -52,6 +52,47 @@ The resulting DOM is essentially:
...
+import fetch from './ApiClient';
+
+function getSignedUrl(file, callback) {
+  const client = new ApiClient();
+  const params = {
+    objectName: file.name,
+    contentType: file.type
+  };
+
+  client.get(signingUrl, { params })
```
Comment (seanadkinson) [inline README.md]: `signingUrl` wouldn't be defined here, so maybe change to `/my/signing/server` or something.

File: README.md [hunk 3]
```diff
@@ -52,6 +52,47 @@ The resulting DOM is essentially:
...
+  getSignedUrl={getSignedUrl}
+  accept="image/*"
+  onProgress={onProgress}
+  onError={onError}
+  onFinish={onFinish}
+  uploadRequestHeaders={{
+    'x-amz-acl': 'public-read'
+  }}
+  contentDisposition="auto"
+  server={apiHost}
```
Comment (seanadkinson) [inline README.md]: Can drop this attribute, since `getSignedUrl` is defined.

**Comment by seanadkinson** [comment]
Looks good, just a couple small things 👍

**Comment by jakubrohleder** [comment]
I've fixed readme :)

File: README.md [hunk 4]
```diff
@@ -52,6 +52,46 @@ The resulting DOM is essentially:
...
+function getSignedUrl(file, callback) {
+  const client = new ApiClient();
+  const params = {
+    objectName: file.name,
+    contentType: file.type
+  };
+
+  client.get('/my/signing/server', { params })
+  .then(data => {
+    callback(data);
```
Comment (seanadkinson) [inline README.md]: Should this be `callback(null, data);`?

Comment (seanadkinson) [inline README.md]: Nevermind, it looks like it doesn't take an error.

**Comment by seanadkinson** [comment]
Pushed `v3.2.0`

**Pull request merged**